hal/src/sercom/i2c.rs

```diff
@@ -293,18 +296,30 @@ pub enum InactiveTimeout {
     Us205 = 0x3,
 }
 
+pub trait I2cMode {}
```

Comment:
I would make this trait sealed so that it can't be implemented outside of this crate:
```
pub trait I2cMode: Sealed {}
```
and also add a short doc comment.